schema_based_on_diagram.sql

```diff
@@ -0,0 +1,47 @@
+-- CREATE DATABASE ACCORDING DIAGRAM
+
+CREATE TABLE patients (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(100),
+  date_of_birth date
+)
+
+CREATE TABLE medical_histories (
+  id SERIAL PRIMARY KEY,
+  admitted_at timestamp,
+  patient_id integer NOT NULL,
+  status VARCHAR(100),
+  FOREIGN KEY (patient_id) REFERENCES patients (id) ON DELETE RESTRICT ON UPDATE CASCADE
+)
+
+CREATE TABLE treatments (
+  id SERIAL PRIMARY KEY,
+  type VARCHAR(100),
+  name VARCHAR(100)
+)
+
+CREATE TABLE invoices (
+  id SERIAL PRIMARY KEY,
+  total_amount decimal,
+  generated_at timestamp,
+  payed_at timestamp,
+  medical_history_id integer NOT NULL,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id) ON DELETE RESTRICT ON UPDATE CASCADE
+)
+
+CREATE TABLE invoice_items (
+  id SERIAL PRIMARY KEY,
+  unit_price decimal,
+  quantity integer,
+  total_price decimal,
+  invoice_id integer NOT NULL,
+  treatment_id integer NOT NULL,
+  FOREIGN KEY (invoice_id) REFERENCES invoices (id) ON DELETE RESTRICT ON UPDATE CASCADE,
+  FOREIGN KEY (treatment_id) REFERENCES treatments (id) ON DELETE RESTRICT ON UPDATE CASCADE
+)
+
```

Comment:
- [ ] As mentioned by the previous reviewer, you are missing a table.  Please create a join table between `medical_histories` and `treatments `named `medical_histories_treaments `.

Look:

![image](https://user-images.githubusercontent.com/57421823/155319215-623bac6b-5c6c-4ed0-8018-a15c6224d8c3.png)

![image](https://user-images.githubusercontent.com/57421823/155319328-fd4e369d-5017-4202-b1d1-da906fd98083.png)

Explanation:  

Many to many relationships in databases require 1 extra table to join the other 2 tables, that new table has references to both tables, thus connecting them.

Hint:

It could include `medical_history_id` and `treatment_id` as columns.

Reply:
Thank you!